core/arch/arm/kernel/user_ta.c

```diff
@@ -657,6 +658,12 @@ static TEE_Result alloc_and_map_ldelf_fobj(struct user_ta_ctx *utc, size_t sz,
 	return res;
 }
 
+static void gdb_helper(const vaddr_t ldelf_addr, const TEE_UUID uuid)
```

Comment:
This looks like it might only only work when compiled with `-O0` or such. I'd prefer something that works regardless of level of optimization.

Reply:
Right, with `03` for example, the symbols aren't there even though the new flag is set to "Y". Any suggestions on how to keep the symbols on over optimization levels? Is there any compiler intrinsic's that would keep the symbols? I've tried to search the gcc documentation, there are a couple of `-fkeep-....` flags, for example
```
-fkeep-static-functions
Emit static functions into the object file, even if the function is never used.
```
That might do the trick?

Reply:
-fkeep-static-functions didn't seem to work ...

```
$ make -j8 CFG_TA_ASLR=y QEMU_VIRTFS_ENABLE=y CFG_GDB_HELPERS=y DEBUG=0 V=1 optee-os
...
/usr/bin//ccache /media/jbech/TSHB_LINUX/devel/optee_projects/qemu/build/../toolchains/aarch32/bin/arm-linux-gnueabihf-gcc -std=gnu99 -fdiagnostics-show-option -Wall -Wcast-align -Werror-implicit-function-declaration -Wextra -Wfloat-equal -Wformat-nonliteral -Wformat-security -Wformat=2 -Winit-self -Wmissing-declarations -Wmissing-format-attribute -Wmissing-include-dirs -Wmissing-noreturn -Wmissing-prototypes -Wnested-externs -Wpointer-arith -Wshadow -Wstrict-prototypes -Wswitch-default -Wwrite-strings -Wno-missing-field-initializers -Wno-format-zero-length -Waggregate-return -Wredundant-decls -Wold-style-definition -Wstrict-aliasing=2 -Wundef -Os -ffunction-sections -fdata-sections -pipe -g3 -fpie -mcpu=cortex-a15 -mfloat-abi=soft -funwind-tables -mthumb -fno-short-enums -fno-common -mno-unaligned-access -fkeep-static-functions -MD -MF out/arm/core/arch/arm/kernel/.user_ta.o.d -MT out/arm/core/arch/arm/kernel/user_ta.o -nostdinc -isystem /media/jbech/TSHB_LINUX/devel/optee_projects/reference/toolchains/aarch32/bin/../lib/gcc/arm-linux-gnueabihf/8.3.0/include -Ilib/libutils/isoc/include -Ilib/libutils/ext/include -Ilib/libmbedtls/include -Ilib/libmbedtls/mbedtls/include -Icore/lib/libtomcrypt/include -Icore/lib/libtomcrypt/src/headers -Icore/lib/libfdt/include -Icore/arch/arm/plat-vexpress/. -Iout/arm/core/arch/arm/kernel -D__KERNEL__ -Icore/include -include out/arm/include/generated/conf.h -Iout/arm/core/include -Icore/arch/arm/include -DARM32=1 -D__ILP32__=1 -DTRACE_LEVEL=3 -Ildelf/include -Ilib/libutee/include -D__FILE_ID__=core_arch_arm_kernel_user_ta_c -c core/arch/arm/kernel/user_ta.c -o out/arm/core/arch/arm/kernel/user_ta.o
...
$ find ../optee_os/out/ -name "*.mk" | xargs grep -e "O0\|O1\|O2\|O3\|Os"                                                                                                                                                                                                     
../optee_os/out/arm/export-ta_arm32/mk/conf.mk:ta_arm32-platform-cflags := -mcpu=cortex-a15 -Os -g3 -fpic -mthumb -fno-short-enums -fno-common -mno-unaligned-access -mfloat-abi=hard -funsafe-math-optimizations -funwind-tables                                             


$ find ../optee_os/ -name "*.elf" | xargs nm -a | grep gdb                                                                                                                                                                                                                    
nm: ../optee_os/out/arm/ta/avb/023f8f1a-292a-432b-8fc4-de8471358067.stripped.elf: no symbols
nm: ../optee_os/out/arm/ta/trusted_keys/f04a0fe7-1f5d-4b9b-abf7-619b85b4ce8c.stripped.elf: no symbols
nm: ../optee_os/out/arm/ta/pkcs11/fd02c9da-306c-48c7-a49c-bbd827ae86ee.stripped.elf: no symbols

```